gitlab_runners inventory plugin: runtime en var and list default behavior

kalw opened the pull request:
Ability to pick options values from env vars on gitlab_runners inventory plugin
Remove default 20 items limit to runners list on gitlab_runners inventory plugin

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Remove default 20 items limit to runners list by adding "all" bool to True in gl.runners.all ref
Permit env var input for server_url and api_token options values
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
plugins/inventory/gitlab_runners.py
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

felixfontein commented on plugins/inventory/gitlab_runners.py:
```diff
@@ -30,11 +30,15 @@
         server_url:
             description: The URL of the GitLab server, with protocol (i.e. http or https).
+            env:
+              - name: GITLAB_SERVER_URL
```
```suggestion
              - name: GITLAB_SERVER_URL
                version_added: 1.0.0
```
Same for the other.

felixfontein commented on plugins/inventory/gitlab_runners.py:
```diff
@@ -95,7 +100,8 @@ def _populate(self):
             else:
-                runners = gl.runners.all()
+                runners = gl.runners.all(all=True)
+                version_added: 1.0.0
```
```suggestion
```
It needs to be added for the `api_token` ini option, not randomly inside the code!

felixfontein commented:
What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

kalw commented:
> What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

👀

kalw commented:
> What you have been adding are not changelog fragments, but `version_added`s. You still need a changelog fragment. (Check the docs I linked for information on how they look like.)

thanks for your patience, should be good :)

felixfontein commented on changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+  - plugin inventory gitlab_runners - Remove default 20 items limit to runners list by adding "all" bool to True 
+  in gl.runners.all (https://github.com/python-gitlab/python-gitlab/blob/df485a92b713a0f2f983c72d9d41ea3a771abf88/gitlab/v4/objects.py#L4922)
+  Permit env var input for server_url and api_token options values
```
```suggestion
  - gitlab_runners inventory plugin - make sure that all runners are returned, and not only the first 20 items (https://github.com/ansible-collections/community.general/pull/611).
minor_changes:
  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611). values.
```

felixfontein commented:
CC @Lunik @SamyCoenen @Shaps @dj-wasabi @marwatk @waheedi - you are maintainers for the gitlab_runner module, so maybe you also have an opinion (or can comment) on this PR for the gitlab_runners inventory plugin.

felixfontein commented:
I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

kalw commented:
> I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

sure, but at least warning this defaults to a subset of the object

Andersson007 commented on changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+  - gitlab_runners inventory plugin - make sure that all runners are returned, and not only the first 20 items (https://github.com/ansible-collections/community.general/pull/611).
+minor_changes:
+  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611). values.
```
```suggestion
  - gitlab_runners inventory plugin - permit environment variable input for ``server_url`` and ``api_token`` options (https://github.com/ansible-collections/community.general/pull/611).
```

kalw commented:
> I just saw #440 - that is probably relevant here as well. Maybe the `all=True` change should be behind a switch?

removed default filter and added os env var capability for filter

Andersson007 commented on changelogs/fragments/611-gitlab-runners-env-vars-intput-and-default-item-limit.yaml:
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - gitlab_runners inventory plugin - permit environment variable input for ``server_url``, ``api_token`` and ``filter`` options (https://github.com/ansible-collections/community.general/pull/611).
```
```suggestion
  - gitlab_runners - permit environment variable input for ``server_url``, ``api_token`` and ``filter`` options (https://github.com/ansible-collections/community.general/pull/611).
```
I fixed that example of fragment a couple of days ago.

Andersson007 replied:
ok, there's no reason to wait

Andersson007 commented:
I just emailed the initial author, let's wait for him for a day

morph027 commented:
Looks good to me.

Andersson007 commented:
@kalw thanks for the contribution!
@felixfontein @morph027 thanks for reviewing!
Merged https://github.com/ansible-collections/community.general/pull/611 into main
